pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -150,10 +150,36 @@ func validateRule(rule *admissionregistration.Rule, fldPath *field.Path, allowSu
 	return allErrors
 }
 
+func isAcceptedAdmissionReviewVersion(v string) bool {
+	return "v1beta1" == v
+}
+
+func validateAdmissionReviewVersions(versions []string, fldPath *field.Path) field.ErrorList {
```

Comment:
this function is still getting called on update (`*Strategy#ValidateUpdate` -> `ValidateValidatingWebhookConfiguration` -> `validateAdmissionReviewVersions`), but lacks the information to know whether it should require a recognized version

we should limit `*Strategy#ValidateUpdate` to calling `Validate*Update`, and not have the strategy also call the `Validate*` method directly (that would let us unit test `Validate*` and `Validate*Update` methods and have that actually reflect what validation would be done on create and update)

Reply:
ValidateUpdate should not call Validate*. If you check current code, both methods it calls doing the same thing and we basically duplicate validation errors.

`strategy#ValidateUpdate` calls to both `ValidateValidatingWebhookConfiguration` and `ValidateValidatingWebhookConfigurationUpdate`. `ValidateValidatingWebhookConfigurationUpdate` calls `ValidateValidatingWebhookConfiguration` on the newObject which is the same as first call `strategy#ValidateUpdate` did to `ValidateValidatingWebhookConfiguration`. I will fix strategy.